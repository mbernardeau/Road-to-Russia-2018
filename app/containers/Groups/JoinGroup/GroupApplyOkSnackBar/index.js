import GroupApplyOkSnackBar from './GroupApplyOkSnackBar';

export default GroupApplyOkSnackBar;
