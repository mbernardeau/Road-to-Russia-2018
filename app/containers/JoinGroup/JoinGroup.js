import React, { Component } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';

import AutoComplete from 'material-ui/AutoComplete';
import RaisedButton from 'material-ui/RaisedButton';

const dataSourceConfig = {
  text: 'name',
  value: 'id',
};

const createDataSource = (groups) => _.map(groups, (value, key) => ({
  [dataSourceConfig.text]: value.name,
  [dataSourceConfig.value]: key,
}));


class JoinGroup extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selectionValid: false,
    };
  }

  handleNewRequest = (chosenRequest, index) => {
    if (index !== -1) {
      this.setState({
        selectionValid: true,
        selected: chosenRequest,
      });
    } else if (this.state.selectionValid) {
      this.setState({
        selectionValid: false,
      });
    }
  }

  applyInGroup = () => {
    if (this.state.selectionValid) {
      this.props.applyInGroup(this.props.uid, this.state.selected);
    }
  }

  render() {
    const groups = createDataSource(this.props.groups);

    return (
      <div style={styles.container}>
        <h2>Rejoindre un groupe</h2>
        <div style={styles.fields}>
          <AutoComplete
            floatingLabelText="Nom du groupe"
            dataSource={groups}
            filter={AutoComplete.fuzzyFilter}
            dataSourceConfig={dataSourceConfig}
            onNewRequest={this.handleNewRequest}
          />
          <RaisedButton disabled={!this.state.selectionValid} onClick={this.applyInGroup} label="Envoyer la demande" primary />
        </div>
      </div>
    );
  }
}

const styles = {
  container: {
    textAlign: 'center',
  },

  fields: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
  },
};

JoinGroup.propTypes = {
  groups: PropTypes.objectOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
    })
  ),
  applyInGroup: PropTypes.func.isRequired,
  uid: PropTypes.string.isRequired,
};

export default JoinGroup;
