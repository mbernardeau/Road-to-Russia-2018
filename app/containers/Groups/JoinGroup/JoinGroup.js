import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { map } from 'lodash';

import Button from 'material-ui/Button';
import Card, { CardActions, CardContent } from 'material-ui/Card';
import { MenuItem } from 'material-ui/Menu';
import Select from 'material-ui/Select';
import { FormControl, FormHelperText } from 'material-ui/Form';
import Typography from 'material-ui/Typography';

import GroupApplyOkSnackBar from './GroupApplyOkSnackBar';

class JoinGroup extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selected: '',
    };
  }

  handleSelection = (event) => {
    this.setState({
      selected: event.target.value,
    });
  }

  handleRequestClose = () => {
    this.setState({
      sent: false,
    });
  }

  applyInGroup = () => {
    if (this.state.selected) {
      this.props.applyInGroup(this.props.uid, this.state.selected);
      this.setState({ selected: '', sent: true });
    }
  }

  render() {
    const { groups, disabled } = this.props;

    return (
      <Card style={styles.fields}>

        <Typography gutterBottom type="headline">Rejoindre un groupe</Typography>
        <br />
        <Typography gutterBottom type="subheading">
          Rejoignez un groupe pour vous confrontez à vos amis, collègues, familles...
        </Typography>

        <CardContent style={styles.content}>
          <FormControl disabled={disabled} fullWidth>
            <Select
              value={this.state.selected}
              onChange={this.handleSelection}
              fullWidth
            >
              {map(groups, (value, key) =>
                <MenuItem key={key} value={key}>{value.name}</MenuItem>
              )}
            </Select>
          </FormControl>
          <FormHelperText>
            {disabled ?
              'Aucun groupe à rejoindre' :
              'Rechercher un groupe'
            }
          </FormHelperText>
        </CardContent>

        <CardActions>
          <Button
            style={styles.button}
            disabled={!this.state.selected}
            onClick={this.applyInGroup}
            color="primary"
            raised
          >
            Envoyer la demande
          </Button>
        </CardActions>

        <GroupApplyOkSnackBar
          open={this.state.sent}
          handleRequestClose={this.handleRequestClose}
        />

      </Card>
    );
  }
}

const styles = {
  fields: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    width: '90%',
    marginBottom: 15,
    padding: 15,
  },

  content: {
    width: '90%',
  },
};

JoinGroup.propTypes = {
  groups: PropTypes.objectOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
    })
  ),
  disabled: PropTypes.bool,
  applyInGroup: PropTypes.func.isRequired,
  uid: PropTypes.string.isRequired,
};

export default JoinGroup;
