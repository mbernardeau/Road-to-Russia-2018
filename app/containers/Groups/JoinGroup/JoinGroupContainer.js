import {
  firebaseConnect,
} from 'react-redux-firebase';
import { connect } from 'react-redux';
import { compose } from 'redux';

import JoinGroup from './JoinGroup';

export default compose(
  firebaseConnect([
    { path: 'groups' },
  ]),
  connect(
    ({ firebase: { data: { groups }, auth: { uid } } }) => ({
      groups,
      uid,
    }),
    (dispatch, { firebase }) => ({
      applyInGroup: (uid, groupId) => firebase.set(`users/${uid}/groups/${groupId}`, true),
    }),
  ),
)(JoinGroup);
