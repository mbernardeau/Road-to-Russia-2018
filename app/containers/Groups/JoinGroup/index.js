import JoinGroup from './JoinGroupContainer';

export default JoinGroup;
