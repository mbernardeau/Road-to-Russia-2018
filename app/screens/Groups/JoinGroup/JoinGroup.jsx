import React, { Component } from 'react'
import PropTypes from 'prop-types'

import Button from 'material-ui/Button'
import Card from 'material-ui/Card'
import CardActions from 'material-ui/Card/CardActions'
import CardContent from 'material-ui/Card/CardContent'
import TextField from 'material-ui/TextField'
import FormControl from 'material-ui/Form/FormControl'
import Typography from 'material-ui/Typography'

import GroupApplyStatus from './GroupApplyStatus'

import './JoinGroup.scss'

class JoinGroup extends Component {
  constructor(props) {
    super(props)
    this.state = {
      code: '',
    }
  }

  handleSelection = event => {
    this.setState({
      code: event.target.value,
    })
  }

  applyInGroup = () => {
    if (this.state.code) {
      this.props.applyInGroup(this.state.code)
      this.setState({ code: '' })
    }
  }

  render() {
    const { code } = this.state

    return (
      <Card className="join-group-card">
        <Typography gutterBottom variant="headline">
          Rejoindre une tribu
        </Typography>
        <br />
        <Typography gutterBottom variant="subheading">
          Rejoignez une tribu pour vous confrontez à vos amis, collègues, familles...
        </Typography>

        <CardContent>
          <Typography gutterBottom variant="caption">
            Entrez le code qui vous a été fourni par l&apos;administrateur de la tribu
          </Typography>
          <FormControl>
            <TextField
              required
              id="join-group-code"
              label="Code"
              value={code}
              margin="normal"
              onChange={this.handleSelection}
            />
          </FormControl>
        </CardContent>

        <CardActions>
          <Button
            disabled={!this.state.code}
            onClick={this.applyInGroup}
            color="primary"
            variant="raised"
          >
            Envoyer la demande
          </Button>
        </CardActions>

        <GroupApplyStatus />
      </Card>
    )
  }
}

JoinGroup.propTypes = {
  applyInGroup: PropTypes.func.isRequired,
}

export default JoinGroup
