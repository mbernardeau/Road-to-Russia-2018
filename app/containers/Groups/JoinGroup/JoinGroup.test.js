import React from 'react';
import renderer from 'react-test-renderer';

import JoinGroup from './JoinGroup';

describe('containers/JoinGroup', () => {
  const props = {
    groups: {
      AA: {
        name: 'group 1',
      },
      BB: {
        name: 'group 2',
      },
      CC: {
        name: 'group 3',
      },
    },
    applyInGroup: () => {},
    uid: 'userId',
  };

  it('renders correctly', () => {
    const tree = renderer.create(
      <JoinGroup {...props} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when disabled', () => {
    const tree = renderer.create(
      <JoinGroup {...props} disabled />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
