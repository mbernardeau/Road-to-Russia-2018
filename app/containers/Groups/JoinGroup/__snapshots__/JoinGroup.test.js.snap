// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/JoinGroup renders correctly 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-shadow2-5 MuiPaper-rounded-2"
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "marginBottom": 15,
      "width": "90%",
    }
  }
>
  <h2>
    Rejoindre un groupe
  </h2>
  <div
    className="MuiCardContent-root-28"
    style={
      Object {
        "width": "90%",
      }
    }
  >
    <div
      className="MuiFormControl-root-29 MuiFormControl-fullWidth-32"
      onBlur={[Function]}
      onFocus={[Function]}
      required={false}
    >
      <div
        className="MuiInput-root-38 MuiInput-fullWidth-52 MuiInput-formControl-39 MuiInput-inkbar-40 MuiInput-underline-46"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          className="MuiSelect-root-33"
        >
          <div
            aria-haspopup="true"
            aria-owns={null}
            aria-pressed="false"
            className="MuiSelect-select-34 MuiSelect-selectMenu-35 MuiInput-input-42 MuiInput-inputSingleline-49"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={undefined}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          />
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            defaultValue={undefined}
            id={undefined}
            name={undefined}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="hidden"
            value=""
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-53 MuiSelect-icon-37"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <p
      className="MuiFormHelperText-root-58"
    >
      Rechercher un groupe
    </p>
  </div>
  <div
    className="MuiCardActions-root-62"
  >
    <button
      className="MuiButtonBase-root-78 MuiButtonBase-disabled-79 MuiButton-root-64 MuiButton-raised-71 MuiButton-raisedPrimary-73 MuiButton-disabled-76 MuiCardActions-actionSpacing-63"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      role="button"
      style={undefined}
      tabIndex={-1}
      type="button"
    >
      <span
        className="MuiButton-label-66"
      >
        Envoyer la demande
      </span>
    </button>
  </div>
</div>
`;

exports[`containers/JoinGroup renders correctly when disabled 1`] = `
<div
  className="MuiPaper-root-1 MuiPaper-shadow2-5 MuiPaper-rounded-2"
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "marginBottom": 15,
      "width": "90%",
    }
  }
>
  <h2>
    Rejoindre un groupe
  </h2>
  <div
    className="MuiCardContent-root-28"
    style={
      Object {
        "width": "90%",
      }
    }
  >
    <div
      className="MuiFormControl-root-29 MuiFormControl-fullWidth-32"
      onBlur={[Function]}
      onFocus={[Function]}
      required={false}
    >
      <div
        className="MuiInput-root-38 MuiInput-disabled-44 MuiInput-fullWidth-52 MuiInput-formControl-39 MuiInput-inkbar-40 MuiInput-underline-46"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        <div
          className="MuiSelect-root-33"
        >
          <div
            aria-haspopup="true"
            aria-owns={null}
            aria-pressed="false"
            className="MuiSelect-select-34 MuiSelect-selectMenu-35 MuiSelect-disabled-36 MuiInput-input-42 MuiInput-inputDisabled-48 MuiInput-inputSingleline-49"
            onBlur={[Function]}
            onClick={null}
            onFocus={undefined}
            onKeyDown={[Function]}
            role="button"
            tabIndex={null}
          />
          <input
            autoComplete={undefined}
            autoFocus={undefined}
            defaultValue={undefined}
            id={undefined}
            name={undefined}
            onKeyDown={undefined}
            onKeyUp={undefined}
            placeholder={undefined}
            readOnly={undefined}
            required={undefined}
            rows={undefined}
            type="hidden"
            value=""
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-53 MuiSelect-icon-37"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
    <p
      className="MuiFormHelperText-root-58"
    >
      Rechercher un groupe
    </p>
  </div>
  <div
    className="MuiCardActions-root-62"
  >
    <button
      className="MuiButtonBase-root-78 MuiButtonBase-disabled-79 MuiButton-root-64 MuiButton-raised-71 MuiButton-raisedPrimary-73 MuiButton-disabled-76 MuiCardActions-actionSpacing-63"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      role="button"
      style={undefined}
      tabIndex={-1}
      type="button"
    >
      <span
        className="MuiButton-label-66"
      >
        Envoyer la demande
      </span>
    </button>
  </div>
</div>
`;
